terraform/modules/forest_node/user-data.tpl

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+NEW_USER="${NEW_USER}"
+USER_DIR="${USER_DIR}"
+USER_DIR_AUTH="${USER_DIR_AUTH}"
+LOKI_ENDPOINT="${LOKI_ENDPOINT}"
+VOLUME_NAME="${VOLUME_NAME}"
+
+# Check if user exists
+if id -u "${NEW_USER}" >/dev/null 2>&1; then
+  echo "User ${NEW_USER} already exists. Exiting..."
+  exit 1
+fi
+
+# Create a login user
+useradd -m ${NEW_USER}
+if [ $? -ne 0 ]; then
+  echo "Failed to create user ${NEW_USER}. Exiting..."
+  exit 1
+fi
+
+# Create .ssh directory for created user
+mkdir -p ${USER_DIR}
+chown ${NEW_USER}:${NEW_USER} ${USER_DIR}
```

Comment:
you should quote variables - can you run this through `shellcheck` please, it should pick up on stuff like this. Alternatively, use the shellcheck extension for vscode